Monthly report August 2025 (#29)
cardano-scaling/website: monthly/2025-08-monthly.md

---
title: August 2025
slug: 2025-08
authors: [jpraynaud, noonio, ch1bo]
tags: [monthly]
---

> This is a monthly report on the progress of 🐲 Hydra and 🛡 Mithril projects since August 2025. It serves as preparation for, and a written summary of, the monthly stakeholder review meeting. The meeting is announced on our Discord channels and held on Google Meet. This month, the meeting took place on August 27, 2025, using these [slides][slides], and you can see the [recording here][recording].

Reviewer: ```suggestion
This is a monthly report on the progress of 🐲 Hydra and 🛡 Mithril projects since August 2025. It serves as preparation for, and a written summary of, the monthly stakeholder review meeting. The meeting is announced on our Discord channels and held on Google Meet. This month, the meeting took place on August 27, 2025, using the slides provided [here][slides], and the recording is available [here][recording].
```

## Mithril

[Issues and pull requests closed in August](https://github.com/input-output-hk/mithril/issues?q=is%3Aclosed+sort%3Aupdated-desc+closed%3A2025-08-01..2025-08-31)

### Roadmap

Below are the latest updates on our roadmap:

- **DMQ signature diffusion prototype** [#2402](https://github.com/input-output-hk/mithril/issues/2402). We have implemented a end to end test of the DMQ protocol for the Mithril nodes, and the networking team has kept implementing the mini-protocols in the Haskell DMQ node.
- **Simple Aggregation Halo2 - Prototype** [#2526](https://github.com/input-output-hk/mithril/issues/2526). We have worked with the innovation team on the implementation of a prototype for a SNARK circuit for Mithril certificates based on Halo2.

> - **Refactor STM library** [#2567](https://github.com/input-output-hk/mithril/issues/2567). We have kept working on the second phase of the refactoring of the STM library.

Reviewer: ```suggestion
- **Refactor STM library** [#2567](https://github.com/input-output-hk/mithril/issues/2567). We continued the second phase of the STM library refactoring.
```

| ------------------ | :--------: | :----------------: | :--------------------: | :---------------: | :------------------: |
| N2C Submission     |     ✓      |         ✓          |           -            |  ✓<sup>\*</sup>   |    _In progress_     |
| N2C Notification   |     ✓      |      Planned       |           ✓            |  ✓<sup>\*</sup>   |    _In progress_     |
| N2N Diffusion      |  Planned   |         -          |           -            |         -         |    _In progress_     |

<i><sup>\*</sup>: for testing purpose only</i>

The network team has kept implementing the DMQ mini-protocols in the Haskell DMQ node. In the mean time, we have implemented a fake DMQ node which allowed us to test the Mithril nodes implementation of the DMQ protocol in an end to end test. This fake DMQ node is implemented in the Mithril relay which now exposes the server sides of the n2c submission and notification mini-protocols. Once the Haskell DMQ node is ready, we will replace the fake DMQ node with the real implementation.

### Prototyping SNARK circuit for Mithril certificates

> We have been working on creating a Mithril certificate with a multi-signature that meets specific requirements for on-chain deployment. The current concatenation proof approach does not meet our constraints, as it exceeds 150 kB with full stake distribution of approximately 2,000 individual signatures, which is too large for on-chain storage (we need < 16 kB).

Reviewer: ```suggestion
We worked on creating a Mithril certificate with a multi-signature that meets specific requirements for on-chain deployment. The current concatenation proof approach does not meet these constraints, as it exceeds 150 kB with a full stake distribution of approximately 2,000 individual signatures. This size is too large for on-chain storage (our target is < 16 kB).
```

### Protocol status

The protocol operated smoothly on the `release-mainnet` network with the following metrics:

- **Registered stake**: `4.7B₳` (`22%` of the Cardano network)
- **Registered SPOs**: `249` (`9%` of the Cardano network)
- **Full Cardano database restorations**: `710` restorations
- **Signer software adoption**: `82.9%` of the SPOs are running a recent version (one of the last three releases).

You can find more information on the [Mithril protocol insights dashboard](https://lookerstudio.google.com/s/mbL23-8gibI).

## Hydra

[Issues and pull requests closed in August](https://github.com/cardano-scaling/hydra/issues?q=is%3Aclosed+sort%3Aupdated-desc+closed%3A2025-08-01..2025-08-31)

![The roadmap with features and ideas](./img/hydra-roadmap-2025-06.png)
<small><center>Snapshot of the new [roadmap](https://github.com/orgs/cardano-scaling/projects/7/views/6) with features and ideas</center></small>

This month, notable [roadmap](https://github.com/orgs/cardano-scaling/projects/7/views/6) updates include:

### [0.22.4 Release](https://github.com/cardano-scaling/hydra/releases/tag/0.22.4)

> This is an important release on top of 0.22.0; featuring many important fixes

Reviewer: ```suggestion
This release builds on 0.22.0 and includes many important fixes
```

observed while testing Hydra for a large operational usecase. In particular we
have now:

- Fixed the API not correctly dealing with log rotation
- Reduced message spam in the presence of mirror nodes
- Fix a bug with an internal queue causing a deadlock
- Fixed an issue with etcd lease renewal
- Implemeneted a workaround for a blocking bug observed with etcd
- Fixed a bug where the hydra-node could stall after a restart (during `ReplayingState`)

> - Drop transactions that could lead to a stuck head

Reviewer: ```suggestion
- Dropped transactions that could lead to a stuck head.
```

### Partial Ada commits

> Not yet released; but on master you can now commit part of the UTxO input:

Reviewer: ```suggestion
Not yet released, but on `master` you can now commit part of the UTXO input:
```

[#2140](https://github.com/cardano-scaling/hydra/issues/2140). Thanks to
aniadev for the great feature request!

### Hydra Doom at RareEvo (again!)

[hydra-doom](https://github.com/cardano-scaling/hydra-doom) returned to Rare Evo 2025. This time running on hydra 0.22.4 on a public network (preprod).

This ran on an ASUS NUC 14 using an N97 chip and 16GB of RAM, allowing us to run both the hydra head and a full preprod node directly on the device using

> the UTxO-HD on-disk backend from cardano-node.

Reviewer: ```suggestion

```
